### PR TITLE
check filenane permissions for possible viewer access issue

### PR DESCRIPTION
**Pull request by bf31415**
check filename permissions for possible view access issue

**Clearly describe the problem and solution**
it's possible that when one ingests a capture but doesn't set the --copy flag, that permissions on the path to the capture flag might not be accessible when viewer runs as dropUser and dropGroup.  the purpose of this code change is to generate a warning if path permission issues might be present which would result in the viewer not being able to open the capture file.

**Relevant issue number(s) if applicable**
1088

**Did you run `npm run lint` from the viewer or parliament directory (whichever you are making changes to) and correct any errors**
not relevant

**Did you Ensure that all tests still pass by navigating to the `tests` directory and running `./tests.pl --viewer`**
ran ./tests.pl and all tests passed

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: capture/reader-libpcap-file.c [hunk 1]
```diff
@@ -179,6 +182,9 @@ LOCAL void reader_libpcapfile_init_monitor()
 LOCAL int reader_libpcapfile_process(char *filename)
 {
     char         errbuf[1024];
+    char 				 path [256];
```
Comment (awick) [inline capture/reader-libpcap-file.c]: Lets use PATH_MAX, and style nic pick path should line up with errbuf

File: capture/reader-libpcap-file.c [hunk 2]
```diff
@@ -179,6 +182,9 @@ LOCAL void reader_libpcapfile_init_monitor()
...
+    char         *p;
+    struct 			 stat stats;
```
Comment (awick) [inline capture/reader-libpcap-file.c]: style nic pick, please change to "struct stat    stats", where the stats lines up with the p

File: capture/reader-libpcap-file.c [hunk 3]
```diff
@@ -196,6 +202,51 @@ LOCAL int reader_libpcapfile_process(char *filename)
         return 1;
     }
 
+    // check to see if viewer might have access issues to non-copied pcap file
+    if (config.copyPcap == 0) {
+
+      if (strlen (filename) > 255) {
```
Comment (awick) [inline capture/reader-libpcap-file.c]: PATH_MAX

File: capture/reader-libpcap-file.c [hunk 4]
```diff
@@ -196,6 +202,51 @@ LOCAL int reader_libpcapfile_process(char *filename)
...
+        // filename bigger than path buffer, skip check
+      } else if ((config.dropUser == NULL) && (config.dropGroup == NULL)) {
+        // drop.User,Group not defined -- skip check
+      } else if (strncmp (filename, "/", 1) != 0) {
+        LOG("WARNING using a relative path may make pcap inaccessible to viewer");
+      } else {
+
+    		path[0] = 0;
+
+        p = strtok (filename, "/");
```
Comment (awick) [inline capture/reader-libpcap-file.c]: need to use strtok_r

Comment (bf31415) [reply capture/reader-libpcap-file.c]: thanks for comments.  making changes.  i also realized I had these two vars defined in the middle of the function-- do you want the definitions at the top at the function level?

          if (stat(path, &stats) != -1) {
            struct group *gr = getgrgid (stats.st_gid);
            struct passwd *pw = getpwuid (stats.st_uid);

Comment (awick) [reply capture/reader-libpcap-file.c]: nope that is fine there, other than the gr/pw should be lined up. :)

**Pull request merged**